tsup.config.base.ts

```diff
@@ -0,0 +1,14 @@
+import type { Options } from "tsup";
+
+export const treeShakableConfig: Options = {
+    splitting: true,
+    clean: true,
+    dts: true,
+    format: ["esm", "cjs"],
+    bundle: true,
+    skipNodeModulesBundle: true,
+    watch: false,
+    outDir: "dist",
+    entry: ["src/**/*.(ts|tsx)", "!src/**/*.test.(ts|tsx)"],
+    shims: true,
```

Comment:
have we agreed at crossmint to ship node.js env shims? Do we def need them or could we force us to use browser-only code and libraries to reduce bundle size?

Reply:
no agreement, i think the bundle size impact is pretty negligible tho and i'd rather not have to think about the compatibility

the docs are not very clear on what exactly it does https://tsup.egoist.dev/#inject-cjs-and-esm-shims